pkg/m3admin/namespace/namespace.go

```diff
@@ -62,39 +68,84 @@ func RequestFromSpec(ns myspec.Namespace) (*admin.NamespaceAddRequest, error) {
 		return nil, fmt.Errorf("preset '%s' not found", ns.Preset)
 	}
 
+	requestOpts, err := requestOptsFromAPI(&opts)
+	if err != nil {
+		return nil, err
+	}
+
 	return &admin.NamespaceAddRequest{
 		Name:    ns.Name,
-		Options: requestOptsFromAPI(&opts),
+		Options: requestOpts,
 	}, nil
 }
 
-func requestOptsFromAPI(opts *myspec.NamespaceOptions) *m3ns.NamespaceOptions {
+func requestOptsFromAPI(opts *myspec.NamespaceOptions) (*m3ns.NamespaceOptions, error) {
+	retentionOpts, err := retentionOptsFromAPI(opts.RetentionOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	indexOpts, err := indexOptsFromAPI(opts.IndexOptions)
+	if err != nil {
+		return nil, err
+	}
+
 	return &m3ns.NamespaceOptions{
 		BootstrapEnabled:  opts.BootstrapEnabled,
 		FlushEnabled:      opts.FlushEnabled,
 		WritesToCommitLog: opts.WritesToCommitLog,
 		CleanupEnabled:    opts.CleanupEnabled,
 		RepairEnabled:     opts.RepairEnabled,
-		RetentionOptions:  retentionOptsFromAPI(opts.RetentionOptions),
+		RetentionOptions:  retentionOpts,
 		SnapshotEnabled:   opts.SnapshotEnabled,
-		IndexOptions:      indexOptsFromAPI(opts.IndexOptions),
-	}
+		IndexOptions:      indexOpts,
+	}, nil
 }
 
-func retentionOptsFromAPI(opts myspec.RetentionOptions) *m3ns.RetentionOptions {
+func retentionOptsFromAPI(opts myspec.RetentionOptions) (*m3ns.RetentionOptions, error) {
```

Comment:
nit: should this be validateRetentionOpts?

Reply:
Hmm I don't quite feel that it's validating, because it's taking one representation (our API) of retention options and transforming it into a new representation (the M3DB API). Maybe `transformRetentionOpts`? But then it's not super clear what's being transformed. Curious your thoughts

Reply:
yea this is very nitty, when reading the invocation above I wasn't sure what this function did. The reason I mentioned validating is due to the transforming of the values to ensure that they are parsed correctly i.e. the error handling. I think `transformRetentionOpts` is not bad or perhaps `normalizeRetentionOpts`? again super nitty and not a big deal in the end.

Reply:
Yeah good call, updated